### PR TITLE
[DDS-1343] Init bay_platform_dependencies module

### PR DESCRIPTION
**Pull request by GROwen**
**Jira issue** https://digital-vic.atlassian.net/browse/DDS-1343

### Test scenario
[dpc-sdp/sdp-ansible-upgrade](https://github.com/dpc-sdp/sdp-ansible-upgrade/tree/feature/DDS-1343--init-bay_platform_deps): branch with customisations for testing. Once this dpc-sdp/baywatch PR is merged a new dpc-sdp/sdp-ansible-upgrade release branch will need to be created referencing the updated Baywatch version.

The above dpc-sdp/sdp-ansible-upgrade branch generated this PR -> https://github.com/dpc-sdp/content-reference-sdp-vic-gov-au/pull/227 which passed CI checks.

### Changes
1. Moves Bay platform specific configuration into a new module.

### CODE REVIEW

File: baywatch.info.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies:
   - smtp:smtp
   - config_split:config_split
   - jwt:jwt
+  - bay_platform_dependencies:bay_platform_dependencies
```
Comment (nicksantamaria) [inline baywatch.info.yml:26]: Do we need the call to `$this->baywatch_install_module('bay_platform_dependencies');` when the dependency is added here?

Comment (GROwen) [reply baywatch.info.yml:26]: [This one](https://github.com/dpc-sdp/baywatch/pull/79/files#diff-15f33d16d9cd8e8b1e20709068a9414d6a238b1a6e9d2f066dee5250a0dd1abbR46)?

Comment (GROwen) [reply baywatch.info.yml:26]: Yeah you do for existing installs.

**Review by nicksantamaria — APPROVED** [review]

**Review by krakerag — APPROVED** [review]
Looks ok to me @GROwen, couple of notes:

- We are actively working on iterating `section_purge` to d10 so that will come with an update to `bay_platform_dependencies` at some stage
- I would look into D10 but the dependencies on this aren't probably ready (see above) so it's probably not the time to do so.

Otherwise all good

**Comment by nicksantamaria** [comment]
@krakerag we're trying to keep the scope of this PR to splitting out platform dependencies from baywatch. I think d10 compatibility is best handled in another PR by an actual Drupal dev.

**Pull request merged**